### PR TITLE
Update encrypt.en.md

### PR DESCRIPTION
**Pull request by Swastyy**
For subtask of #19241 

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: docs/document/content/dev-manual/encrypt.en.md [hunk 1]
```diff
@@ -5,20 +5,22 @@ weight = 12
...
-## SPI Interface
+## EncryptAlgorithm
 
-| SPI Name         | Description            |
-| ---------------- | ---------------------- |
-| EncryptAlgorithm | Data encrypt algorithm |
+### Fully-qualified class name
 
-## Sample
+[`org.apache.shardingsphere.encrypt.algorithm`](https://github.com/apache/shardingsphere/blob/master/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/spi/EncryptAlgorithm.java)
```
Comment (terrymanu) [inline docs/document/content/dev-manual/encrypt.en.md:12]: Only package name is not enough, please add class name

File: docs/document/content/dev-manual/encrypt.en.md [hunk 2]
```diff
@@ -5,20 +5,22 @@ weight = 12
...
-| RC4EncryptAlgorithm           | RC4 data encrypt algorithm |
-| SM3EncryptAlgorithm           | SM3 data encrypt algorithm |
-| SM4EncryptAlgorithm           | SM4 data encrypt algorithm |
+Data encryption algorithm definition
+
+### Implementation classes
+
+| *Configuration Type* | *Description*                             | *Fully-qualified class name* |
+| -------------------- | ----------------------------------------- | ---------------------------- |
+| MD5EncryptAlgorithm           | MD5 data encrypt algorithm | [`org.apache.shardingsphere.encryption.algorithm.MD5Encrypt`](https://github.com/apache/shardingsphere/blob/master/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/MD5EncryptAlgorithm.java) |
```
Comment (terrymanu) [inline docs/document/content/dev-manual/encrypt.en.md:22]: MD5EncryptAlgorithm should be MD5, please reference the `getType` of source code. Other rows are same

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: docs/document/content/dev-manual/encrypt.en.md (hunk 2 above)
Comment (Swastyy) [reply docs/document/content/dev-manual/encrypt.en.md]: done

File: docs/document/content/dev-manual/encrypt.en.md (hunk 1 above)
Comment (Swastyy) [reply docs/document/content/dev-manual/encrypt.en.md:12]: how to get that?

Comment (terrymanu) [reply docs/document/content/dev-manual/encrypt.en.md:12]: For this case, it should be `org.apache.shardingsphere.encrypt.spi.EncryptAlgorithm`

File: docs/document/content/dev-manual/encrypt.en.md (hunk 2 above)
Comment (terrymanu) [reply docs/document/content/dev-manual/encrypt.en.md]: Great

**Review by terrymanu — APPROVED** [review]

**Pull request merged**